[OSDOCS#17174]: Make changes as per the errors and warnings for CQA-1 (#114085)
openshift/openshift-docs: modules/compliance-crd-advanced-compliance-scan.adoc

[id="advance-compliance-scan-object_{context}"]
= Advanced ComplianceScan Object

[role="_abstract"]

> The `ComplianceScan` object provides advanced options such as custom result storage, debug pods, and scan suspension for troubleshooting and tool integration.

Reviewer: We are supposed to make the short description about what the user can do, not what the feature does. 
See the guidelines: [Avoid feature-focused language (“This product allows you to…”)](https://docs.google.com/document/d/1xRc_licMTbLrLG9vmJ4bP9aPk6mzwwly5lGlkRz9d9s/edit?tab=t.0).

```suggestion
You can use the `ComplianceScan` object to configure advanced options such as custom result storage, debug pods, and scan suspension for troubleshooting and tool integration.
```

> The Compliance Operator includes options for advanced users for debugging or integrating with existing tool. While it is recommended that you should not create a `ComplianceScan` object directly, you can instead manage it by using a `ComplianceSuite` object.

Reviewer 2: ```suggestion
The Compliance Operator includes options for advanced users for debugging or integrating with existing tool. While Red Hat recommends that you should not create a `ComplianceScan` object directly, you can instead manage the object by using a `ComplianceSuite` object.
```

"it" can be a vague referrent.

metadata:
  name: <name_of_the_compliance_scan>
spec:
  scanType: Node
  profile: xccdf_org.ssgproject.content_profile_moderate
  content: ssg-ocp4-ds.xml
  contentImage: registry.redhat.io/compliance/openshift-compliance-content-rhel8@sha256:45dc...
  rule: "xccdf_org.ssgproject.content_rule_no_netrc_files"
  nodeSelector:
    node-role.kubernetes.io/worker: ""
status:
  phase: DONE
  result: NON-COMPLIANT
----
where:

`spec.scanType`:: Specifies either `Node` or `Platform`. Node profiles scan the cluster nodes and platform profiles scan the Kubernetes platform.
`spec.profile`:: Specifies the XCCDF (Extensible Configuration Checklist Description Format) identifier of the profile that you want to run.
`spec.contentImage`:: Specifies the container image that encapsulates the profile files. 
`spec.rule`:: Specifies the scan to run a single rule. This rule has to be identified with the XCCDF ID, and has to belong to the specified profile. If you skip the `rule` parameter, the scan runs for all the available rules of the specified profile.

> `spec.nodeSelector`:: If you are on the {product-title} and wants to generate a remediation, the `nodeSelector` label has to match the `MachineConfigPool` label. If you do not specify the `nodeSelector` parameter or match the `MachineConfig` label, the scan still runs but a remediation is not created.

Reviewer 3: ```suggestion
`spec.nodeSelector`:: If you are on the {product-title} and wants to generate a remediation, the `nodeSelector` label must match the `MachineConfigPool` label. If you do not specify the `nodeSelector` parameter or match the `MachineConfig` label, the scan still runs but a remediation is not created.
```

metadata:
  name: <name_of_the_compliance_scan>
spec:
  scanType: Node
  profile: xccdf_org.ssgproject.content_profile_moderate
  content: ssg-ocp4-ds.xml
  contentImage: registry.redhat.io/compliance/openshift-compliance-content-rhel8@sha256:45dc...
  rule: "xccdf_org.ssgproject.content_rule_no_netrc_files"
  nodeSelector:
    node-role.kubernetes.io/worker: ""
status:
  phase: DONE
  result: NON-COMPLIANT
----
+

> where:

Reviewer 2: Dangling plus:

<img width="857" height="314" alt="Image" src="https://github.com/user-attachments/assets/8f390d44-a836-49e2-b9fd-1fe93309b164" />

`spec.scanType`:: Specifies either `Node` or `Platform`. Node profiles scan the cluster nodes and platform profiles scan the Kubernetes platform.

> `spec.profile`:: Specifies the XCCDF identifier of the profile that you want to run.

Reviewer 2: XCCDF expand on first use?

`spec.contentImage`:: Specifies the container image that encapsulates the profile files. 

> `spec.rule`:: Specifies the scan to run a single rule. This rule has to be identified with the XCCDF ID, and has to belong to the specified profile. If you skip the `rule` parameter, then scan runs for all the available rules of the specified profile.

Reviewer 2: ```suggestion
`spec.rule`:: Specifies the scan to run a single rule. This rule has to be identified with the XCCDF ID, and has to belong to the specified profile. If you skip the `rule` parameter, the scan runs for all the available rules of the specified profile.
```

> `spec.nodeSelector`:: If you are on the {product-title} and wants to generate a remediation, then nodeSelector label has to match the `MachineConfigPool` label. If you do not specify `nodeSelector` parameter or match the `MachineConfig` label, scan will still run, but it will not create remediation.

Reviewer 2: ```suggestion
`spec.nodeSelector`:: If you are on the {product-title} and wants to generate a remediation, the `nodeSelector` label has to match the `MachineConfigPool` label. If you do not specify the `nodeSelector` parameter or match the `MachineConfig` label, the scan still runs but a remediation is not created.
```

Author: Changed as per the feedback

    node-role.kubernetes.io/worker: ""
status:
  phase: DONE
  result: NON-COMPLIANT
----
where:

`spec.scanType`:: Specifies either `Node` or `Platform`. Node profiles scan the cluster nodes and platform profiles scan the Kubernetes platform.
`spec.profile`:: Specifies the XCCDF (Extensible Configuration Checklist Description Format) identifier of the profile that you want to run.
`spec.contentImage`:: Specifies the container image that encapsulates the profile files. 
`spec.rule`:: Specifies the scan to run a single rule. This rule must be identified with the XCCDF ID, and must belong to the specified profile. 
+
[NOTE]
====
If you skip the `rule` parameter, then scan runs for all the available rules of the specified profile.
====
`spec.nodeSelector`:: If you are on the {product-title} and wants to generate a remediation, the `nodeSelector` label must match the `MachineConfigPool` label. If you do not specify the `nodeSelector` parameter or match the `MachineConfig` label, the scan still runs but a remediation is not created.
+
[NOTE]
====
If you do not specify `nodeSelector` parameter or match the `MachineConfig` label, scan will still run, but it will not create remediation.

> ====

Reviewer: You can probably leave this as a note only. 

```suggestion
`spec.nodeSelector`:: If you are on the {product-title} and wants to generate a remediation, the `nodeSelector` label must match the `MachineConfigPool` label. 
+
[NOTE]
====
If you do not specify `nodeSelector` parameter or match the `MachineConfig` label, scan will still run, but it will not create remediation.
====
```
